gt/main.c

```diff
@@ -702,8 +811,35 @@ config_gt_instance(struct gt_config *gt_conf, unsigned int lcore_id)
 		goto free_lua_state;
 	}
 
+	if (gt_conf->configured_proto & GT_CONFIGURED_IPV4) {
+		ret = setup_neighbor_tbl(
+			rte_lcore_to_socket_id(gt_conf->lcores[0]),
+			lcore_id * RTE_MAX_LCORE + 0, ETHER_TYPE_IPv4,
+			(1 << (32 - gt_conf->ip4_addr_plen)),
+			&instance->neigh);
+		if (ret < 0)
+			goto free_lua_state;
+	}
+
+	if (gt_conf->configured_proto & GT_CONFIGURED_IPV6) {
+		ret = setup_neighbor_tbl(
+			rte_lcore_to_socket_id(gt_conf->lcores[0]),
+			lcore_id * RTE_MAX_LCORE + 1, ETHER_TYPE_IPv6,
+			gt_conf->max_num_ipv6_neighbors, &instance->neigh6);
+		if (ret < 0)
+			goto free_neigh;
+	}
+
 	goto out;
 
+free_neigh:
+	if (!(gt_conf->configured_proto & GT_CONFIGURED_IPV4))
+		goto free_lua_state;
+
+	rte_free(instance->neigh.cache_tbl);
+	instance->neigh.cache_tbl = NULL;
+	rte_hash_free(instance->neigh.hash_table);
+	instance->neigh.hash_table = NULL;
```

Comment:
Is it possible to replace the labels `free_neigh` and `free_lua_state` with a simple call to `cleanup_gt_instance()`?